### ISSUE TITLE
GD2 pod went into Crashbackloop mode , brick is killed of same pod

### ISSUE THREAD

**Issue by ksandha** [gluster/glusterd2#1343]
1) Create gcs cluster
2) create a pvc 
3) create an app pod  claiming the pvc
4) touch 2 files on the mount point.
5) kill a brick from a gd2 and touch same file names again and create a new file 
```
<<gd2 pod >>
[root@gluster-kube2-0 brick]# 
[root@gluster-kube2-0 brick]# 
[root@gluster-kube2-0 brick]# kill -9 49
[root@gluster-kube2-0 brick]# 
<<app pod>>
root@redis1:/mnt/gluster1# touch ab c
root@redis1:/mnt/gluster1# touch ab c
root@redis1:/mnt/gluster1# touch cde 
```

6) check heal info 
```
[root@gluster-kube2-0 brick]# glustercli volume heal info pvc-624c4d1b-f16c-11e8-bbd1-52540066b71b
Brick: gluster-kube2-0.glusterd2.gcs:/var/run/glusterd2/bricks/pvc-624c4d1b-f16c-11e8-bbd1-52540066b71b/subvol1/brick1/brick
Status: Transport endpoint is not connected
entries: -1

Brick: gluster-kube1-0.glusterd2.gcs:/var/run/glusterd2/bricks/pvc-624c4d1b-f16c-11e8-bbd1-52540066b71b/subvol1/brick2/brick
Status: Connected
entries: 4
b2ce2b7f-0f45-4e61-b424-6e115cf45c60:/ab
099482bb-c8ba-4dc5-95c4-5befc014dd08:/c
1c032981-f721-4de9-83e3-9a91ad83d986:/cde
00000000-0000-0000-0000-000000000001:/

Brick: gluster-kube3-0.glusterd2.gcs:/var/run/glusterd2/bricks/pvc-624c4d1b-f16c-11e8-bbd1-52540066b71b/subvol1/brick3/brick
Status: Connected
entries: 4
b2ce2b7f-0f45-4e61-b424-6e115cf45c60:/ab
099482bb-c8ba-4dc5-95c4-5befc014dd08:/c
1c032981-f721-4de9-83e3-9a91ad83d986:/cde
00000000-0000-0000-0000-000000000001:/

[root@gluster-kube2-0 brick]# 

```

7) Check the status of the cluster:-
```
[vagrant@kube1 ~]$ kubectl get pods -n gcs
NAME                                   READY   STATUS             RESTARTS   AGE
csi-attacher-glusterfsplugin-0         2/2     Running            0          111m
csi-nodeplugin-glusterfsplugin-pr249   2/2     Running            0          111m
csi-nodeplugin-glusterfsplugin-smt2j   2/2     Running            0          111m
csi-nodeplugin-glusterfsplugin-vx9dz   2/2     Running            0          111m
csi-provisioner-glusterfsplugin-0      3/3     Running            1          111m
etcd-8zpggh6hjf                        1/1     Running            0          116m
etcd-bbhclh7rs2                        1/1     Running            0          117m
etcd-jwcbng4pp5                        0/1     Running            0          40m
etcd-operator-7cb5bd459b-4g6d9         1/1     Running            0          120m
gluster-kube1-0                        1/1     Running            1          115m
gluster-kube2-0                        0/1     CrashLoopBackOff   8          39m
gluster-kube3-0                        1/1     Running            1          115m

```
### Observed behavior
gd2 got into crashloopback mode while checking the heal info. the 

### Expected/desired behavior
gd2 pods should be working fine, even in between gd2 commands. 

### Information about the environment:

- Glusterd2 version used (e.g. v4.1.0 or master): 
[root@gluster-kube1-0 /]# rpm -qa | grep gluster
glusterfs-client-xlators-6dev-0.285.git4249783.el7.x86_64
glusterfs-api-6dev-0.285.git4249783.el7.x86_64
glusterfs-fuse-6dev-0.285.git4249783.el7.x86_64
glusterfs-server-6dev-0.285.git4249783.el7.x86_64
glusterd2-5.0-0.dev.39.git65dafcd.el7.x86_64
glusterfs-libs-6dev-0.285.git4249783.el7.x86_64
glusterfs-6dev-0.285.git4249783.el7.x86_64
glusterfs-cli-6dev-0.285.git4249783.el7.x86_64
gluster-prometheus-exporter-0.3-dev.26.git8107970.el7.x86_64

- Operating system used: 
- Glusterd2 compiled from sources, as a package (rpm/deb), or container: 
container
- Using External ETCD: (yes/no, if yes ETCD version):
external
- If container, which container image: 
- Using kubernetes, openshift, or direct install: 
kubernetes
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: 

### Other useful information

- glusterd2 config files from all nodes (default /etc/glusterd2/glusterd2.toml)
- glusterd2 log files from all nodes (default /var/log/glusterd2/glusterd2.log)
- ETCD configuration
- Contents of `uuid.toml` from all nodes (default /var/lib/glusterd2/uuid.toml)
- Output of `statedump` from any one of the node

hit this issue 1/1
### Useful commands

- To get glusterd2 version
    ```
[root@gluster-kube1-0 /]# glusterd2 --version
glusterd version: v6.0-dev.39.git65dafcd
git SHA: 65dafcd
go version: go1.9.4
go OS/arch: linux/amd64

    ```
- To get ETCD version
    ```
    etcd --version
    ```
- To get output of statedump
    ```
    curl http://glusterd2-IP:glusterd2-Port/statedump
    ```

**Comment by Madhu-1** [comment]
I think the issue is not because of checking the healing status, it's because of killing the brick causes huge memory utilization by the gd2 pod.

related to 
https://github.com/gluster/gcs/issues/71
https://github.com/gluster/glusterd2/issues/1321

I feel this is kind of duplicate issue

**Comment by Madhu-1** [comment]
Fixed in #1321
